app/pages/SearchResultsPage/SearchResultsPage.tsx

```diff
@@ -34,22 +36,64 @@ interface SearchState extends ParsedQs {
 
 /** Wrapper component that handles state management, URL parsing, and external API requests. */
 export const SearchResultsPage = () => {
+  const [cookies] = useCookies(["googtrans"]);
+  const history = useHistory();
+  const { search } = useLocation();
   const { userLocation } = useAppContext();
   const [lastPush, setLastPush] = useState(Date.now());
-  const { search } = useLocation();
   const [expandList, setExpandList] = useState(false);
-
-  const searchState: SearchState = useMemo(
-    () => qs.parse(search.slice(1)),
-    [search]
-  );
+  const [searchState, setSearchState] = useState<SearchState>({});
   const [searchRadius, setSearchRadius] = useState(
     searchState?.configure?.aroundRadius ?? "all"
   );
+  // In cases where we translate a non-English query into English (to query Alglolia), we use this
+  // state var to cache the user's orginal, untranslated query to display on the results page
+  const [untranslatedQuery, setUntranslatedQuery] = useState("");
+  const [searchStateResolved, setSearchStateResolved] = useState(false);
+
+  useEffect(() => {
+    const queryParams = qs.parse(search.slice(1));
+    const { query } = queryParams;
+    setUntranslatedQuery(query as string);
+
+    // Google Translate determines translation source and target with a
+    // "googtrans" cookie. If the cookie exists, we assume that the
+    // the query should be translated into English prior to querying Algolia
+    const translationCookie = cookies.googtrans;
+
+    if (query && translationCookie) {
+      const [, targetLanguage] = translationCookie.split("/en/") as string;
+      if (targetLanguage !== "en") {
+        post("/api/translation/translate_text", {
+          text: query,
+          target_language: targetLanguage,
+        }).then((resp) =>
+          resp.json().then((body) => {
+            setSearchState({
+              ...queryParams,
+              query: body.result,
+            });
+            setSearchStateResolved(true);
+          })
+        );
+      }
```

Comment:
Don't we need an `else` to go with this inner `if`? Otherwise, wouldn't `searchStateResolved` never get set to `true`, hanging the page?

Reply:
Ooof, good catch